internal/sign/job/signer.go

```diff
@@ -182,6 +190,22 @@ func (s *signer) getHashAnnotationValue(ctx context.Context, privateSecret, publ
 	return getHashValue(podTemplate, publicKeyData, privateKeyData)
 }
 
+func (s *signer) getSAImageRepoSecret(ctx context.Context, mod *kmmv1beta1.Module, accountName string) *v1.LocalObjectReference {
+	serviceaccount := v1.ServiceAccount{}
+
+	namespacedName := types.NamespacedName{Name: accountName, Namespace: mod.Namespace}
+
+	err := s.client.Get(ctx, namespacedName, &serviceaccount)
+	if err != nil {
+		return nil
+	}
+
+	if len(serviceaccount.Secrets) == 0 {
+		return nil
+	}
+	return &v1.LocalObjectReference{Name: serviceaccount.Secrets[0].Name}
```

Comment:
Don't we want to return the whole slice and mount that?

Reply:
Annoyingly we get ObjectReference's and we need LocalObjectReference's for the secrets so this is a hoop that casts these.  However I was assuming that the serviceaccount only had one secret, so I've changed this so it returns the serviceaccount.Secrets array and we loop through that to create out volumes (casting as we go)

I will add removing the casts to my list of future improvements but it needs changes to functions other used in other places so will need a wider PR.